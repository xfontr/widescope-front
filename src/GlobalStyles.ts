import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  body {
    margin: 0;
    min-height: 100vh;
  }

  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  button, input {
    font: inherit;
    border: none;
  }
`;

export default GlobalStyles;
