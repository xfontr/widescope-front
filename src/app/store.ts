import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import { userDataReducer } from "../store/slices/userData/userDataSlice";
import { userReducer } from "../store/slices/user/userSlice";

export const store = configureStore({
  reducer: {
    user: userReducer,
    userData: userDataReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
