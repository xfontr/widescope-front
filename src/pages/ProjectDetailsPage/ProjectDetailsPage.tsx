import { memo, useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import ProjectDetails from "../../components/ProjectDetails/ProjectDetails";
import useProjects from "../../hooks/useProjects/useProjects";
import { IProject } from "../../types/project";

const ProjectDetailsPage = memo((): JSX.Element => {
  const { getById } = useProjects();
  const { projectId } = useParams();
  const [currentProject, setProject] = useState(false as IProject | false);

  useEffect(() => {
    (async () => {
      try {
        const retreivedProject = await getById(projectId as string);
        setProject(retreivedProject as IProject);
      } catch (error) {}
    })();
  }, [getById, projectId, setProject]);

  return (
    <>
      {!currentProject && (
        <div className="page-container">
          <span>Project not found.</span>
        </div>
      )}
      {currentProject && <ProjectDetails project={currentProject} />}
    </>
  );
});

export default ProjectDetailsPage;
