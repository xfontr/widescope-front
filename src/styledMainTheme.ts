const styledMainTheme = {
  colors: {
    primary: "rgb(255, 255, 255)",
    primaryDarker: "rgb(238, 238, 238)",

    secondary: "rgb(25, 29, 35)",
    secondaryBrigther: "rgba(25, 29, 35, 0.7)",
  },

  shapes: {
    radiusSmall: "0.75rem",
    radiusBig: "1.5rem",

    shortShadow: "6px 6px 18px 6px rgba(0, 0, 0, 0.1)",
    longShadow: "10px 10px 30px 10px rgba(0, 0, 0, 0.1)",

    rectangularContainer: "0.8rem 1.5rem 0.8rem 1.5rem",
  },

  spacing: {
    gapSmall: "1rem",
    gapBig: "1.5rem",

    paddingSmall: "0.8rem",
    paddingBig: "1.5rem",
  },

  breakpoints: {
    small: "390px",
    big: "700pxpx",
    large: "1050px",
  },
};

export default styledMainTheme;
