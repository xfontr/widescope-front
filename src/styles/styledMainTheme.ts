const styledMainTheme = {
  colors: {
    primary: "rgb(255, 255, 255)",
    primaryDarker: "rgb(238, 238, 238)",

    secondary: "rgb(25, 29, 35)",
    secondaryBrigther: "rgba(25, 29, 35, 0.7)",

    error: "rgb(179, 120, 120)",
    success: "green",
  },

  shapes: {
    radiusSmall: "0.4rem",
    radiusBig: "1.5rem",

    shortShadow: "6px 6px 18px 6px rgba(0, 0, 0, 0.1)",
    longShadow: "10px 10px 30px 10px rgba(0, 0, 0, 0.1)",

    rectangularContainer: "0.7rem 1.6rem",
  },

  spacing: {
    gapSmall: "1.7rem",
    gapBig: "2.5rem",

    paddingSmall: "0.8rem",
    paddingBig: "1.8rem",
  },

  breakpoints: {
    small: "390px",
    big: "700px",
    large: "1250px",
  },
};

export default styledMainTheme;
