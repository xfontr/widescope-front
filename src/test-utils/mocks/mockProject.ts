import { IProject } from "../../types/project";

const mockProject: IProject = {
  id: "630e31ecb968115ba6a3e29f",
  name: "Project123",
  description: "The description",
  technologies: ["React", "Express"],
  repository: "www.google.com",
  author: "username",
  authorId: "id",
  logo: "ProjectLogo.png",
  logoBackup: "BackupProjectLogo.png",
  creationDate: "1662627402346",
};

export default mockProject;
