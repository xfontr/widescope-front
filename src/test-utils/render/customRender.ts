import { render } from "@testing-library/react";
import { Wrapper } from "./Wrapper";

const customRender = (view: JSX.Element, ...options: any) =>
  render(view, { wrapper: Wrapper, ...options });

export * from "@testing-library/react";

export { customRender as render };
