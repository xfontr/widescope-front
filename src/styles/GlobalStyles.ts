import { createGlobalStyle } from "styled-components";
import styledMainTheme from "./styledMainTheme";

const GlobalStyles = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  body {
    margin: 0;
    min-height: 100vh;
    background-color: ${styledMainTheme.colors.primaryDarker};
  }

  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  button, input {
    font: inherit;
    border: none;
  }
`;

export default GlobalStyles;
