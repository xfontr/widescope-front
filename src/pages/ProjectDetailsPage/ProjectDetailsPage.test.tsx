import { act } from "react-dom/test-utils";
import mockProject from "../../test-utils/mocks/mockProject";
import { render, screen } from "../../test-utils/render/customRender";
import ProjectDetailsPage from "./ProjectDetailsPage";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../app/store";
import { ThemeProvider } from "styled-components";
import styledMainTheme from "../../styles/styledMainTheme";
import { BrowserRouter } from "react-router-dom";
import GlobalStyles from "../../styles/GlobalStyles";

let mockProjectId = mockProject.id;

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({ projectId: mockProjectId }),
}));

describe("Given a ProjectDetailsPage component", () => {
  describe("When instantiated with a valid param", () => {
    test("Then it should match the snapshot", () => {
      const expectedPage = renderer.create(
        <Provider store={store}>
          <ThemeProvider theme={styledMainTheme}>
            <BrowserRouter>
              <GlobalStyles />
              <ProjectDetailsPage />
            </BrowserRouter>
          </ThemeProvider>
        </Provider>
      );

      expect(expectedPage).toMatchSnapshot();
    });

    test("Then it should get a project and display it", async () => {
      await render(<ProjectDetailsPage />);

      const projectTitle = await screen.findByRole("heading", {
        name: mockProject.name,
      });

      expect(projectTitle).toBeInTheDocument();
    });
  });

  describe("When instantiated with an invalid param", () => {
    test("Then it should display a not found message", async () => {
      mockProjectId = "falseId";

      // eslint-disable-next-line testing-library/no-unnecessary-act
      await act(() => {
        render(<ProjectDetailsPage />);
      });

      const projectTitle = screen.queryByRole("heading", {
        name: mockProject.name,
      });

      const notFoundMessage = screen.getByText("Project not found.");

      expect(projectTitle).not.toBeInTheDocument();
      expect(notFoundMessage).toBeInTheDocument();
    });
  });
});
