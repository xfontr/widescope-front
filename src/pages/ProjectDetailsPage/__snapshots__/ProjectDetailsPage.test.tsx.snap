// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ProjectDetailsPage component When instantiated with a valid param Then it should match the snapshot 1`] = `
<div
  className="page-container"
>
  <span>
    Project not found.
  </span>
</div>
`;
